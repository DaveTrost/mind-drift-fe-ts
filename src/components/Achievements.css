@import './vars.css';
.AchievementsContainer {
  ul {
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-bottom: 6em;
    align-items: center;
  }
  .Achievements {
    li{
      display: flex;
      align-items: center;
      width: 85%;
      border: solid var(--light-green) .2em;
      border-radius: .2em;
      margin-bottom: 1em;
      img {
        height: 75px;
      }
      h3 {
        text-align: center;
        font-family: Montserrat;
        font-weight: 700;
        font-size: 1em;
        margin-top: .5em;
        text-transform: uppercase;
        color: var(--rust);
        margin-bottom: .2em;
      }
      p {
        font-family: Montserrat;
        font-weight: 400;
        line-height: normal;
        font-size: .7em;
        color: var(--dark-blue);
        margin: 1em;
      }
    }
  }
}
