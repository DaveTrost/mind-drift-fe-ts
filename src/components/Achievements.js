import React from 'react';
import PropTypes from 'prop-types';
import styles from './Achievements.css';


const Achievements = ({ achieves }) => {
  const mappedAchieve = achieves.map(achievement => {
    return (
      <li key={achievement._id}>
        <img src={achievement.img} />
        <div>
          <h3>{achievement.name}</h3>
          <p>{achievement.description}</p>
        </div>
      </li>
    );
  });
  return (
    <div className={styles.AchievementsContainer}>
      <ul className={styles.Achievements}>
        {mappedAchieve}
      </ul>
    </div>
  );
};

Achievements.propTypes = {
  achieves: PropTypes.array
};

export default Achievements;

