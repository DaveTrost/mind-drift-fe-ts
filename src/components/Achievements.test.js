import React from 'react';
import { shallow } from 'enzyme';
import Achievements from './Achievements';

describe('Achievements component', () => {
  it('renders empty Achievements', () => {
    const wrapper = shallow(<Achievements achieves={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders some achievements', () => {
    const achieves = [
      {
        _id: '1',
        name: 'First Steps',
        img: 'https://raw.githubusercontent.com/mindDrift/mind-drift-fe/dev/src/assets/firstStepsicon.png',
        description: 'Congratulations, you have logged your first breathing session with mindDrift',
      },
      {
        _id: '2',
        name: '2-Day Streak',
        img: 'https://raw.githubusercontent.com/mindDrift/mind-drift-fe/dev/src/assets/streakTWO.png',
        description: 'Congratulations, you completed a breathing session every day for the past 2 days!',
      },
      {
        _id: '3',
        name: '5-Day Streak',
        img: 'https://raw.githubusercontent.com/mindDrift/mind-drift-fe/dev/src/assets/streakFIVE.png',
        description: 'Congratulations, you completed a breathing session every day for the past 2 days!',
      }
    ];
    const wrapper = shallow(<Achievements achieves={achieves} />);
    expect(wrapper).toMatchSnapshot();
  });
});


