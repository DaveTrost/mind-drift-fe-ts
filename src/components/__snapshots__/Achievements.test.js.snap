// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Achievements component renders empty Achievements 1`] = `
<div
  className="AchievementsContainer"
>
  <ul
    className="Achievements"
  />
</div>
`;

exports[`Achievements component renders some achievements 1`] = `
<div
  className="AchievementsContainer"
>
  <ul
    className="Achievements"
  >
    <li
      key="1"
    >
      <img
        src="https://raw.githubusercontent.com/mindDrift/mind-drift-fe/dev/src/assets/firstStepsicon.png"
      />
      <div>
        <h3>
          First Steps
        </h3>
        <p>
          Congratulations, you have logged your first breathing session with mindDrift
        </p>
      </div>
    </li>
    <li
      key="2"
    >
      <img
        src="https://raw.githubusercontent.com/mindDrift/mind-drift-fe/dev/src/assets/streakTWO.png"
      />
      <div>
        <h3>
          2-Day Streak
        </h3>
        <p>
          Congratulations, you completed a breathing session every day for the past 2 days!
        </p>
      </div>
    </li>
    <li
      key="3"
    >
      <img
        src="https://raw.githubusercontent.com/mindDrift/mind-drift-fe/dev/src/assets/streakFIVE.png"
      />
      <div>
        <h3>
          5-Day Streak
        </h3>
        <p>
          Congratulations, you completed a breathing session every day for the past 2 days!
        </p>
      </div>
    </li>
  </ul>
</div>
`;
